### PR TITLE
User orders

### LINKED ISSUE #67: Backend can get product history
- [ ] write tests for GraphQL getUserOrders query 
- [ ] write graphQL query for getUserOrders 
- [ ] getUserOrders will return UserOrderType
- [ ] write resolver for getUserOrder query 
- [ ] resolver will have joins with Crates, (user?), and inner joins with products through productDeliveries 

FE #53

### PR DESCRIPTION
**Pull request by jdcarey128**
## What's this PR do?
- Adds ordersByUser query which returns all productDeliveries and nested products for each order a user has

## Relevant tickets
- closes #67 

## Where should the reviewer start?
- take a look at the test file to see the structure of the data returned 

## How should this be manually tested?
- n/a

## Any background context you want to provide?
- n/a

## Screenshots/GIFs
- 

## Questions:
-

### CODE REVIEW

File: code/api/.gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ node_modules/
 build/
 public/images/uploads/*
 !public/images/uploads/.gitkeep
+./package-lock.json
```
Comment (kncaputo) [inline code/api/.gitignore:5]: AWESOME! We should've done this a long time ago

File: code/api/src/modules/order/resolvers.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export async function getByUser(parentValue, {}, { auth }) {
+  if(auth.user && auth.user.id > 0) {
+    return await models.Order.findAll({ raw: true,
+      where: {
+        userId: auth.user.id
+      },
+      include: [
+        {model: models.User, as: 'user' },
+        {model: models.Crate, as: 'crate' },
+        {model: models.ProductDelivery, as: 'productDeliveries',
```
Comment (CalebCyphers) [inline code/api/src/modules/order/resolvers.js:22]: Awesome, thanks for getting this up and running!

File: code/api/src/modules/order/tests/queries.test.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { request, express, graphqlHTTP, schema, connection } from '../../../setup/test_helper'
+
+import serverConfig from '../../../config/server.json'
+import models from '../../../setup/models'
+import authentication from '../../../setup/authentication'
+
+describe('orders by user', () => {
```
Comment (CalebCyphers) [inline code/api/src/modules/order/tests/queries.test.js:7]: solid testing here

**Review by CalebCyphers — APPROVED** [review]

File: code/api/src/modules/order/types.js [hunk 4]
```diff
@@ -14,12 +15,16 @@ const OrderType = new GraphQLObjectType({
     id: {type: GraphQLInt },
     user: { type: UserType },
     crate: { type: CrateType },
-    productDeliveries: { type: new GraphQLList( ProductDeliveryType )},
+<<<<<<< HEAD
+    productDeliveries: { type: new GraphQLList(ProductDeliveryType) },
+=======
+>>>>>>> Update types and order query
+    products: { type: new GraphQLList(ProductType) },
```
Comment (kncaputo) [inline code/api/src/modules/order/types.js:22]: It looks like there might be some leftover rebase or merge notes in here? @jdcarey128

File: code/api/src/modules/order/types.js [hunk 5]
```diff
@@ -14,12 +15,16 @@ const OrderType = new GraphQLObjectType({
     id: {type: GraphQLInt },
     user: { type: UserType },
     crate: { type: CrateType },
-    productDeliveries: { type: new GraphQLList( ProductDeliveryType )},
+<<<<<<< HEAD
```
Comment (saraho1123) [inline code/api/src/modules/order/types.js:18]: Looks like there is a conflict from a rebase. Probably need to fix this before merge! @jdcarey128 @GarrettCottrell @the-color-bliu

File: code/api/src/modules/productDelivery/types.js [hunk 6]
```diff
@@ -6,13 +6,13 @@ import { ProductType } from '../product/types'
 
 // Product Delivery Type
 const ProductDeliveryType = new GraphQLObjectType({
-  name: 'product delivery',
+  name: 'ProductDelivery',
   description: 'Product Delivery Type',
 
   fields: () => ({
     id: { type: GraphQLInt },
     product: { type: ProductType },
```
Comment (saraho1123) [inline code/api/src/modules/productDelivery/types.js:14]: Is it possible to add a name for the product. So that FE can display that for the user?

Comment (saraho1123) [reply code/api/src/modules/productDelivery/types.js:14]: nvm. I found it in product!

File: code/api/src/modules/user/query.js [hunk 7]
```diff
@@ -47,3 +47,9 @@ export const userGenders = {
   type: new GraphQLList(UserGenderType),
   resolve: getGenders
 }
+
+// Orders
```
Comment (saraho1123) [inline code/api/src/modules/user/query.js:51]: I notice there is commented out code. Are you saving it for future reference?

**Review by kncaputo — CHANGES_REQUESTED** [review]

File: code/api/src/modules/user/types.js [hunk 8]
```diff
@@ -46,13 +46,13 @@ const UserGenderType = new GraphQLObjectType({
...
-  name: 'userOrders',
-  description: 'User Order Type',
-
-  fields: () => ({
-    orders: { type: new GraphQLList( OrderType )}
-  })
-})
-
-export { UserType, UserLoginType, UserGenderType, UserOrdersType }
+// const UserOrdersType = new GraphQLObjectType({
```
Comment (saraho1123) [inline code/api/src/modules/user/types.js:49]: Just want to make a note that there is commented out code. No worries if it is for future reference!

**Review by saraho1123 — COMMENTED** [review]
Awesome! As well as I can tell as a FE, it looks great! There is the one file ordery/types.js that looks like it has an unresolved conflict. Thanks y'all! Looking forward to hooking this up!

**Review by kncaputo — APPROVED** [review]

**Pull request merged**